### PR TITLE
Add metric for general MSAS statistics

### LINKED ISSUE #639: Add metric for general MSAS statistics
### Problem Description
In [this paper](https://arxiv.org/pdf/2207.14406), we introduced a new methodology for calculating multi-sequence metrics called MSAS. We should add the MSAS-related metrics to SDMetrics so that users with sequential data can use them for evaluation.

### Expected behavior
Add a metric called `StatisticMSAS` that performs the MSAS algorithm for a given statistic.

**Data compatibility**: 1 ID column (representing the sequence key), and 1 continuous column (datetime or numerical)

**Parameters**:
- (required) `real_data`: A tuple of 2 pandas.Series objects. The first represents the sequence key of the real data and the second represents a continuous column of data.
- (required) `synthetic_data`: A tuple of 2 pandas.Series objects. The first represents the sequence key of the synthetic data and the second represents a continuous column of data.
- `statistic`: A string representing the statistic function to use when computing MSAS
    - (default) `'mean'`: The arithmetic mean
    - `'median'`: The median value
    - `'std'`: The standard deviation
    - `'min'`: The min value
    - `'max'`: The max value

**Output**: A score in range [0, 1] -- 0 being the worst and 1 being the best

```python
from sdmetrics.column_pairs import StatisticMSAS

score = StatisticMSAS.compute(
  real_data=(real_table['patient_id'], real_table['heart_rate']),
  synthetic_data = (synthetic_table['patient_id'], synthetic_table['heart_rate']),
  statistic='std'
)
```

**How does it work?** The sequence key determines which continuous values belong to which sequence. This metric computes a statistic for all sequences in the real and synthetic data, and then compares those distributions.

1. Calculate the statistic value of each sequence in the real data (call this distribution D_r)
2. Calculate the statistic value of each sequence in the synthetic data (call this distribution D_s)
3. Now apply the [KSComplement metric](https://docs.sdv.dev/sdmetrics/metrics/metrics-glossary/kscomplement) to compare the similarities of the distributions (D_r, D_s). Return this score.

### PR DESCRIPTION
**Pull request by fealho**
CU-86b2evqnt, Resolve #639.

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b2evqnt SDMetrics - Add metric for general MSAS statistics #639](https://app.clickup.com/t/86b2evqnt)

File: sdmetrics/timeseries/statistic_msas.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            'min': np.min,
+            'max': np.max,
+        }
+        if statistic not in statistic_functions:
+            raise ValueError(
+                f'Invalid statistic: {statistic}.'
+                f' Choose from [{", ".join(statistic_functions.keys())}].'
+            )
+
+        real_keys, real_values = real_data
```
Comment (amontanez24) [inline sdmetrics/timeseries/statistic_msas.py:77]: should we check to make sure the inputs are tuples as well before trying to unpack?

File: sdmetrics/timeseries/statistic_msas.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        synthetic_keys, synthetic_values = synthetic_data
+        stat_func = statistic_functions[statistic]
+
+        def calculate_statistics(keys, values):
+            statistics = []
+            for key in keys.unique():
+                group_values = values[keys == key].to_numpy()
+                statistics.append(stat_func(group_values))
+            return pd.Series(statistics)
```
Comment (amontanez24) [inline sdmetrics/timeseries/statistic_msas.py:86]: Is this faster than combining the two series into a dataframe and then using groupby with an [aggregation](https://pandas.pydata.org/docs/user_guide/groupby.html#built-in-aggregation-methods) function?

**Review by amontanez24 — APPROVED** [review]
:shipit:

**Review by frances-h — APPROVED** [review]
LGTM!

**Review by R-Palazzo — APPROVED** [review]

**Pull request merged**